### PR TITLE
FISH-9754 Implemented obfuscation for logs using regex.

### PR DESCRIPTION
**Pull request by NotedSalmon**
(Payara6) I have implemented obfuscation for the logs. The `LogCollector` class calls the `Obfuscation` class which contains the functionality involved with obfuscating files. 

In the `LogCollector` I have updated the constructors to include the `CollectorService` as I needed to pass the boolean variable `obfuscateDomainXml` as to use it as a way to indentify if the obfuscation parameter is set to true which is then used in the if statement in the function `visitFile` which will lead to the function created in `Obfuscation`.

The `obfuscateLogData` function uses a regex to identify different IP Addresses and checks to see if it is equal to the default address which is `0.0.0.0`. It then goes throw the log trying to match a key from the map `hostReplacements` and rewriting the value accordingly.

### CODE REVIEW

File: src/main/java/fish/payara/extras/diagnostics/util/Obfuscation.java [hunk 1]
```diff
@@ -166,13 +174,39 @@ private void obfuscateAttribute(Element element, String attributeName) {
...
+                    obfuscatedReplacement = OBFUSCATED_CHANGE + (hostsReplacements.size() + 1);
                     hostsReplacements.put(keywordAttribute, obfuscatedReplacement);
-                    obfuscatedCounter++;
                 }
                 element.setAttribute(attributeName, obfuscatedReplacement);
             }
         }
     }
 
+    public static void obfuscateLogData(Path sourceFile, Path destinationFile) throws IOException {
```
Comment (kalinchan) [inline src/main/java/fish/payara/extras/diagnostics/util/Obfuscation.java:185]: Is there a reason to be static?

File: src/main/java/fish/payara/extras/diagnostics/util/Obfuscation.java [hunk 2]
```diff
@@ -166,13 +174,39 @@ private void obfuscateAttribute(Element element, String attributeName) {
...
+        try (BufferedReader reader = Files.newBufferedReader(sourceFile);
+             BufferedWriter writer = Files.newBufferedWriter(destinationFile)) {
+            Pattern pattern = Pattern.compile("(([0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])\\.){3}([0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])");
```
Comment (kalinchan) [inline src/main/java/fish/payara/extras/diagnostics/util/Obfuscation.java:188]: You can move this out and make this pattern static

**Comment by jGauravGupta** [comment]
LGTM, Please review @kalinchan points.

**Review by jGauravGupta — APPROVED** [review]

**Review by breakponchito — APPROVED** [review]

File: src/main/java/fish/payara/extras/diagnostics/util/Obfuscation.java (hunk 1 above)
Comment (NotedSalmon) [reply src/main/java/fish/payara/extras/diagnostics/util/Obfuscation.java:185]: If the function was not static, line 165-167 in `LogCollector.java` would not work

`if (obfuscateEnabled){`
             `Obfuscation.obfuscateLogData(file, resolvedDestination);`
 #`}else {`

**Pull request merged**